app/models/grouping.rb

```diff
@@ -25,6 +25,8 @@ class Grouping < ApplicationRecord
   has_many :ta_memberships, class_name: 'TaMembership'
   has_many :tas, through: :ta_memberships, source: :user
   has_many :students, through: :student_memberships, source: :user
+  has_many :test_students, class_name: 'TestStudent', through: :student_memberships, source: :user
```

Comment:
I'm not sure if this is necessary. I don't think the grouping object needs to differentiate between students and test_students here.